components/nimbus/src/behavior.rs

```diff
@@ -292,6 +301,79 @@ impl EventQueryType {
                     values.0 as f64 / values.1 as f64
                 }
             }
+            Self::LastSeen => match buckets.into_iter().position(|v| v > &0) {
+                Some(v) => v as f64,
+                None => f64::MAX,
+            },
+        })
+    }
+
+    pub fn validate_arguments(
+        self,
+        args: &[Value],
+    ) -> Result<(String, Interval, usize, usize, Self)> {
```

Comment:
Actually I think my brain wasn't firing on all cylinders when writing this -- I have access to the original `Self` in the evaluator, because this method is being called on it. I'm going to remove `Self` from the tuple return.